backend/src/Controllers/elections.controllers.ts

```diff
@@ -9,14 +9,44 @@ var ElectionsModel = new ElectionsDB();
 const getElectionByID = async (req: any, res: any, next: any) => {
     console.log(`-> elections.getElectionByID ${req.params.id}`)
     try {
-        const election = await ElectionsModel.getElectionByID(parseInt(req.params.id))
+        var election = await ElectionsModel.getElectionByID(parseInt(req.params.id))
         console.log(`get election ${req.params.id}`)
-        if (!election){
+        if (!election) {
             console.log('Error')
             return res.status('400').json({
                 error: "Election not found"
             })
         }
+        // Update Election State
+        const currentTime = new Date();
+        if (election.state === 'draft') {
+
+        }
+        if (election.state === 'finalized') {
+            if (election.start_time) {
+                console.log(`-> Election Transitioning To Open From ${election.state}`)
+                const startTime = new Date(election.start_time);
+                if (currentTime.getTime() > startTime.getTime()) {
+                    election.state = 'open';
+                    election = await ElectionsModel.updateElection(election)
+                }
+            } else {
+                console.log(`-> Election Transitioning To Open From ${election.state}`)
+                election.state = 'open';
+                election = await ElectionsModel.updateElection(election)
+            }
+        }
+        if (election.state === 'open') {
+            if (election.end_time) {
+                const endTime = new Date(election.end_time);
+                if (currentTime.getTime() > endTime.getTime()) {
+                    console.log('-> Election Transitioning To Closed')
+                    election.state = 'closed';
+                    election = await ElectionsModel.updateElection(election)
```

Comment:
nit: In theory ``updateElection()`` could get called twice if the first vote gets cast after the end period. It could be refactored as follows

```
if(election.state === 'finalized' or election.state === 'open' or election.state === 'closed'){
    if(election.end_time and currentTime.getTime() > endTime.getTime()){
       // become closed 
    }elif(election.start_time and currentTime.getTime() < startTime.getTime()){
      // become finalized
    }else{
      // become open
    }
}
```

Reply:
I think this would result in updating the election every time even if the state doesn't change. I added a check to only run the update once.

Reply:
ah good point

Thanks for adding the updates!